### ISSUE TITLE
Serialize value classes

### ISSUE THREAD

**Issue by gzoller** [salat/salat#81]
Hello,

Today if I have a simple, single-parameter wrapper class:

case class URL( u:String ) {...}

and its used in another class:

case class Foo( myURL : URL ) ...
Foo( URL("something") )

If I serialize this I get { "myURL" : {"u" : "something } } 

Works just fine, but a little messy.
I was thinking about Scala's new value classes.  Because we know they have one parameter we could streamline the serialization to the much-cleaner:

{"myURL" : "something"}

This is a lot cleaner.  I don't know, though, if Scala puts enough info in the class files to allow grater to do its magic like it does w/case classes.  I'm hoping the fact this is a value class isn't used/discarded by the compiler before grater could see it.

Greg

**Comment by mkljakubowski** [comment]
You can easily bypass this issue with a custom Transformer. I wasted a lot of time looking for solution and it's so simple. 
Example is available at: https://github.com/salat/salat/blob/master/salat-core/src/test/scala/com/novus/salat/test/custom/model.scala

**Comment by noahlz** [comment]
Here is a small example that demonstrates this issue:

```
class IntHolder(val i: Int) extends AnyVal 
case class IntHolderHolder(i: IntHolder)
```

and then

```
val intHolder = new IntHolder(1)
val model = IntHolderHolder(intHolder)
val json = grater[IntHolderHolder].toCompactJSON(model) 
```

The resulting error in `1.9.10`

```
com.novus.salat.json.UnsupportedJsonTransformationException: serialize: Unsupported JSON transformation for class='com.novus.salat.examples.IntHold
er', value='com.novus.salat.examples.IntHolder@1'

NOTE: salat has certain limitations. It cannot deserialize JSON into case classes having fields such as:
- Arrays such as Array[String]
- Nested collections such as Map[String, List[String]]
- Optional collections such as Option[List[String]]

For more information, please visit: https://github.com/salat/salat/wiki/SupportedTypes
```

**Comment by noahlz** [comment]
I agree with @mkljakubowski that a custom transformer is the way to go here.  I'm not certain this use case is frequent enough to justify the JVM reflection hoops we'd have to jump through (and corresponding code complexity it would introduce) to support `AnyVal` (type erasure!), when the Transformer code is so straight forward.
